app/routers/salary/utils.py

```diff
@@ -0,0 +1,544 @@
+        ) -> Dict[str, config.NUMERIC]:
+    """Returns all details and calculation for the given year and month based
+    on the provided settings, including specified additions or deductions.
+
+    Args:
+        year (int): The relevant year.
+        month (int): The relevant month.
+        wage (SalarySettings): The relevant salary calculation settings.
+        overtime (bool): If True, monthly overtime is calculated and added to
+                         final salary.
+        bonus (int | float, optional): Amount to be added to the final salary.
+        deduction (int | float, optional): Amount to be subtracted from the
+                                           final salary.
+
+    Returns:
+        dict(str: int | float): All details and calculation for the given year
+                                and month based on the provided settings,
+                                including specified additions or deductions.
+
+    Raises:
+        ValueError: If  year is smaller than 1 or larger than 9999, or if month
+                    is smaller than 1 or larger than 12.
+    """
+    # Code revision required after categories feature is added
+    month_start, month_end = get_month_times(year, month)
+    shifts = get_event_by_category(month_start, month_end, wage.user_id,
+                                   wage.category_id)
+    weeks = get_relevant_weeks(year, month)
+    base_salary = sum(calc_shift_salary(shift.start, shift.end, wage)
+                      for shift in shifts)
+    if overtime:
+        month_weekly_overtime = get_monthly_overtime(shifts, weeks, wage)
+    else:
+        month_weekly_overtime = 0
+    transport = calc_transport(len(shifts), wage.daily_transport)
+    salary = round(sum((base_salary, bonus,
+                       month_weekly_overtime, transport)), 2)
+    if deduction > salary:
+        deduction = salary
+    salary -= deduction
+    return {
+        'year': year,
+        'month': month,
+        'num_of_shifts': len(shifts),
+        'base_salary': base_salary,
+        'month_weekly_overtime': month_weekly_overtime,
+        'transport': transport,
+        'bonus': bonus,
+        'deduction': deduction,
+        'salary': round(salary, 2),
+        }
+
+
+def get_settings(session: Session, user_id: int,
+                 category_id: int) -> Optional[SalarySettings]:
+    """Returns settings for `user_id` and `category_id` if exists, None
+    otherwise.
+
+    Args:
+        session (Session): DB session.
+        user_id (int): Id of the relevant user.
+        category_id (int): Id for the relevant category.
+
+    Returns:
+        SalarySettings | None: Settings for the provided user_id and
+                               category_id if exists, None otherwise.
+    """
+    settings = session.query(SalarySettings).filter(and_(
+        SalarySettings.user_id == user_id, SalarySettings.category_id
+        == category_id)).first()
+    session.close()
```

Comment:
Is this necessary?

Reply:
I can't really say why, but for some reason removing this line fails one of my tests.
The functionality is essentially the same, and the service itself doesn't seem to be affected, yet I didn't manage to pass the tests without it.
If you want to take a look at it, the one that fails is "test_routes: :test_edit_settings".

Reply:
Try to `git checkout upstream/develop -- requirements.txt` and `pip install -U -r requirements.txt`.
We've downgraded the pytest-asyncio library, that's might be the problem

Reply:
Unfortunately this did not solve the problem.